src/formatting/lists.rs

```diff
@@ -486,11 +497,12 @@ where
                 }
                 // An additional space for the missing trailing separator (or
                 // if we skipped alignment above).
-                if !formatting.align_comments
-                    || (last
-                        && item_max_width.is_some()
-                        && !separate
-                        && !formatting.separator.is_empty())
+                if !inner_item.is_empty()
+                    && (!formatting.align_comments
+                        || (last
+                            && item_max_width.is_some()
+                            && !separate
+                            && !formatting.separator.is_empty()))
```

Comment:
I know this one was already pretty busy but the extra condition and subsequent indentation is a bit of an eye sore. I guess it'd be okay but I also wouldn't mind seeing some minor refactoring of this existing code too